v03_pipeline/lib/annotations/fields.py

```diff
@@ -0,0 +1,77 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any
+
+from v03_pipeline.lib.annotations import (
+    gcnv,
+    reference_dataset_collection,
+    sample_lookup_table,
+    shared,
+    snv,
+    sv,
+)
+from v03_pipeline.lib.model import AnnotationType, DatasetType
+
+if TYPE_CHECKING:
+    import hail as hl
+
+ANNOTATION_CONFIG = {
+    (DatasetType.SNV, AnnotationType.FORMATTING): [
+        shared.rg37_locus,
+        shared.sorted_transcript_consequences,
+        shared.variant_id,
+        shared.xpos,
+    ],
+    (DatasetType.SNV, AnnotationType.REFERENCE_DATASET_COLLECTION): [
+        reference_dataset_collection.hgmd,
+        reference_dataset_collection.gnomad_non_coding_constraint,
+        reference_dataset_collection.screen,
+    ],
+    (DatasetType.SNV, AnnotationType.SAMPLE_LOOKUP_TABLE): [
+        sample_lookup_table.AC,
+        sample_lookup_table.AF,
+        sample_lookup_table.AN,
+    ],
+    (DatasetType.SNV, AnnotationType.GENOTYPE_ENTRIES): [
+        snv.gq,
+        snv.ab,
+        snv.dp,
+        shared.sample_id,
+    ],
+    (DatasetType.MITO, AnnotationType.FORMATTING): [
+        shared.rg37_locus,
+        shared.sorted_transcript_consequences,
+        shared.variant_id,
+        shared.xpos,
+    ],
+    (DatasetType.MITO, AnnotationType.SAMPLE_LOOKUP_TABLE): [
+        sample_lookup_table.AC,
+        sample_lookup_table.AF,
+        sample_lookup_table.AN,
+    ],
+    (DatasetType.SV, AnnotationType.FORMATTING): [
+        shared.rg37_locus,
+        sv.variant_id,
+        shared.xpos,
+    ],
+    (DatasetType.GCNV, AnnotationType.FORMATTING): [
+        gcnv.variant_id,
+        gcnv.xpos,
+    ],
+}
+
+
+def get_fields(
+    t: hl.Table | hl.MatrixTable,
+    annotation_type: AnnotationType,
+    **kwargs: Any,
+) -> dict[str, hl.Expression]:
+    dataset_type = kwargs['dataset_type']
+    fields = {
+        field_expression.__name__: field_expression(t, **kwargs)
```

Comment:
Important (and maybe controversial) design choice here to have all of the annotation functions accept all of the task arguments (`DatasetType`, `Env`, `ReferenceGenome`, `liftover_ref_path` etc.).  Only a few of the functions (`rg37_locus`, `hgmd`, and the interval tables) need additional args...  so we're maybe over-designing for the special cases.

Reply:
I think having them passed as keyword args is a reasonable compromise for making the calling logic straightforward, but the functions that do not actually need them can simply pass them through